### DISCUSSION TITLE
Log everything in a file

### DISCUSSION THREAD

**Discussion by Ghdhdhdh** [BattlesnakeOfficial/feedback#261] [Feature Requests]
My data science friend really wants to analyse my snake, but the leaderboard only shows top 100. Pls add. log in a CSV

**Comment by xtagon** [comment]
Hi @Ghdhdhdh,

While the leaderboard doesn't currently have this feature, the solution some people land on is to log games from within your snake's server. Since every game sends the full turn's state to your snake for each move request, it's easy enough to log what was sent just before deciding your move. How to do this depends on what language/framework you're using, and what format you want your logs to be in. For example, the move request is going to be JSON, not CSV, so if you need CSV you'd want to post-process that to extract the fields you want.

The main downside is that you will only receive logs for games that your snake is in. But often that's good enough.
